fix: do not add sse-related compiler flags on architectures that don't support them

doudou opened the pull request.

chhtz commented on CMakeLists.txt:
```diff
@@ -1,7 +1,14 @@
+add_definitions(-Wall -Wno-unused-function -march=native)
+include(CheckCXXCompilerFlag)
+
+check_cxx_compiler_flag("-msse2;-mssse3;-mfpmath=sse" __supports_sse3)
+if (__supports_sse3)
+    add_definitions(-msse2 -mssse3 -mfpmath=sse)
+endif()
+
```
This whole block can be removed (except for the first line). `-march=native` includes all SSE versions that are available. Also `-mfpmath=sse` is standard on x86_64 anyways, even `-march=native` probably does not make that much difference anymore (these manual flags look like a relic from long time ago ...)

doudou replied:
TIL ...

Updated the PR. I added a definition to allow overriding the target arch (since we're building in the cloud for particular machines, march=native might give us some weird results)

doudou committed:
march=native already enables SSE when present, remove explicit setting

This ensures the package can build on arch that do not have SSE
(e.g. arm). Additionally, the patch allows overriding 'native'
via the OPT_TARGET_ARCH option to allow building on a specific CPU
while targetting another.
